src/main/java/com/gluonhq/richtext/action/TextDecorateAction.java

```diff
@@ -30,8 +30,8 @@ protected void bind() {
             if (nv != null && !updating) {
                 updating = true;
                 TextDecoration.Builder builder = TextDecoration.builder();
-                Decoration decoration = viewModel.getDecoration();
-                if (viewModel.getSelection() == Selection.UNDEFINED && decoration instanceof TextDecoration) {
+                Decoration decoration = viewModel.getDecorationAtCaret();
+                if (viewModel.getSelection().isDefined() && decoration instanceof TextDecoration) {
```

Comment:
I am trying to understand why this change is done here?

Reply:
Hmm, I might have missed an `!`?